jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPart.java

```diff
@@ -1339,7 +1339,8 @@ private boolean parseContent(Content.Chunk chunk)
                     }
                     else
                     {
-                        // The boundary was partially matched.
+                        // The boundary was partially matched, but it
```

Comment:
This section is confusing with nested if structure:
```java
if (boudnaryMatch > 0)
{
    if (boundaryMath == X)
    {
        // A
    }
    else
    {
        // B
    }
}
else
{
    // C
}
```
This should be:

```java
if (boudnaryMatch > 0)
{
    // A
} 
else if (boundaryMath == X)
{
    // B
}
else
{
    // C
}
```

Reply:
It is not equivalent.
In your version, when `boundaryMatch == X` it will execute the branch where `boundaryMatch > 0`, which is wrong.

Reply:
Ooops I meant:
```java
if (boudnaryMatch == X)
{
    // A
} 
else if (boundaryMath > 0)
{
    // B
}
else
{
    // C
}
```